source/blender/gpu/intern/gpu_codegen.c

```diff
@@ -1297,7 +1297,13 @@ static void gpu_node_input_link(GPUNode *node, GPUNodeLink *link, const GPUType
 		input->type = type;
 		input->source = GPU_SOURCE_VEC_UNIFORM;
 
-		memcpy(input->vec, link->ptr1, type * sizeof(float));
+		if (type == GPU_INT) {
```

Comment:
We can even make this GPU_INT check only and left all above changes as they are in current master

Reply:
What about using an array for size of types indexed by GPUType ? So you can change the values and avoid exceptions in the code when adding conditions.

Reply:
I'm not convinced to do that before final release. I would prefer fix memcpy and make the tiny refactor after

Reply:
As you want, but the refactor is just defining an array of 18 elements.